Automattic/wp-calypso
# Update Ownerinfo and Other Controls With New `getUserOwnsPurchase` Selector

#### Changes proposed in this Pull Request

* Add new `getUserOwnsPurchase` selector
* Refactor `OwnerInfo` using 👆
* Rename `owner-Info.tsx` to `owner-info.tsx`
* Fix typo `this.renderPurhaseItemContent` in `PurchaseItem`

#### Testing instructions
1. Boot branch
2. Navigate to `/plans/my-plan/:siteSlug` for a site with a Product ( Jetpack Backup or Scan, for example )
3. Verify that the "Manage Subscription" Button is clickable and does not have an info popover with "To manage this subscription [...]" in it. <img width="928" alt="Screen Shot 2022-01-06 at 13 56 18" src="https://user-images.githubusercontent.com/2810519/148457916-3cc42393-3500-4f4a-b370-e675bd0e6932.png">
4. Repeat steps 2 & 3 for a _Plan_ ( any paid WordPress.com plan, Jetpack Security, etc. )
5. Navigate to `/plans/:siteSlug` for a Jetpack Site with any paid plan or product ( make sure to test both a plan and product ).
6. Verify the "Manage Subscription" button is clickable and does not have an info popover with "To manage this subscription [...]" in it. <img width="924" alt="Screen Shot 2022-01-06 at 13 56 48" src="https://user-images.githubusercontent.com/2810519/148457881-a398dd49-c564-49a6-a354-8dff13643ea4.png">
7. Navigate to `/me/purchases`
8. Verify every purchase listed that you own does not have an info popover with "To manage this subscription [...]" next to the product title
<img width="347" alt="Screen Shot 2022-01-06 at 13 52 24" src="https://user-images.githubusercontent.com/2810519/148457423-89c8a3c9-911a-4dd0-b9b6-dfac61946787.png">



Fixes #59820
